go/benchmarks/main.go

```diff
@@ -0,0 +1,184 @@
+
+func main() {
+	opts := parseArguments()
+
+	runConfig, err := verifyOptions(opts)
+	if err != nil {
+		log.Fatal("Error verifying options:", err)
+		return
+	}
+
+	if runConfig.resultsFile != os.Stdout {
+		defer closeFile(runConfig.resultsFile)
+	}
+
+	err = runBenchmarks(runConfig)
+	if err != nil {
+		log.Fatal("Error running benchmarking:", err)
+	}
+}
+
+func closeFile(file *os.File) {
+	err := file.Close()
+	if err != nil {
+		log.Fatal("Error closing file:", err)
+	}
+}
+
+func parseArguments() *options {
+	resultsFile := flag.String("resultsFile", "results/go-results.json", "Result filepath")
+	dataSize := flag.String("dataSize", "[100]", "Data block size")
+	concurrentTasks := flag.String("concurrentTasks", "[1 10 100 1000]", "Number of concurrent tasks")
+	clientNames := flag.String("clients", "all", "One of: all|go-redis|glide")
+	host := flag.String("host", "localhost", "Hostname")
+	port := flag.Int("port", 6379, "Port number")
+	clientCount := flag.String("clientCount", "[1]", "Number of clients to run")
+	tls := flag.Bool("tls", false, "Use TLS")
+	clusterModeEnabled := flag.Bool("clusterModeEnabled", false, "Is cluster mode enabled")
+	minimal := flag.Bool("minimal", false, "Run benchmark in minimal mode")
+
+	flag.Parse()
+
+	return &options{
+		resultsFile:        *resultsFile,
+		dataSize:           *dataSize,
+		concurrentTasks:    *concurrentTasks,
+		clients:            *clientNames,
+		host:               *host,
+		port:               *port,
+		clientCount:        *clientCount,
+		tls:                *tls,
+		clusterModeEnabled: *clusterModeEnabled,
+		minimal:            *minimal,
+	}
+}
+
+func verifyOptions(opts *options) (*runConfiguration, error) {
+	var runConfig runConfiguration
+	var err error
+
+	if opts.resultsFile == "" {
+		runConfig.resultsFile = os.Stdout
+	} else {
+		err = os.MkdirAll(filepath.Dir(opts.resultsFile), os.ModePerm)
+		if err != nil {
+			return nil, err
+		}
+
+		runConfig.resultsFile, err = os.Create(opts.resultsFile)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	runConfig.concurrentTasks, err = parseOptionsIntList(opts.concurrentTasks)
+	if err != nil {
+		return nil, fmt.Errorf("invalid concurrentTasks option: %v", err)
+	}
+
+	runConfig.dataSize, err = parseOptionsIntList(opts.dataSize)
+	if err != nil {
+		return nil, fmt.Errorf("invalid dataSize option: %v", err)
+	}
+
+	runConfig.clientCount, err = parseOptionsIntList(opts.clientCount)
+	if err != nil {
+		return nil, fmt.Errorf("invalid clientCount option: %v", err)
+	}
+
+	switch {
+	case strings.EqualFold(opts.clients, goRedis):
+		runConfig.clientNames = append(runConfig.clientNames, goRedis)
+
+	case strings.EqualFold(opts.clients, glide):
+		runConfig.clientNames = append(runConfig.clientNames, glide)
+
+	case strings.EqualFold(opts.clients, all):
+		runConfig.clientNames = append(runConfig.clientNames, goRedis, glide)
+	default:
+		return nil, fmt.Errorf("invalid clients option, should be one of: all|go-redis|glide")
+	}
```

Comment:
Same here about indentation. I guess that's just how switch works? Also kind of confused about which variable this is switching on.

Reply:
yeah that's the way the indentation works with Go. This switches on the conditions specified by each case rather than on a variable. This construct also shows up in Go's examples:
https://go.dev/tour/flowcontrol/11